src/uct/cuda/cuda_ipc/cuda_ipc_cache.c

```diff
@@ -108,14 +109,57 @@ static void uct_cuda_ipc_cache_invalidate_regions(uct_cuda_ipc_cache_t *cache,
               cache->name, from, to);
 }
 
-ucs_status_t uct_cuda_ipc_unmap_memhandle(void *rem_cache, uintptr_t d_bptr,
+static
+ucs_status_t uct_cuda_ipc_get_rem_cache(uct_cuda_ipc_iface_t *iface, pid_t pid,
+                                        uct_cuda_ipc_cache_t **cache)
+{
+    char target_name[64];
+    khiter_t khiter;
+    int khret;
+    ucs_status_t status;
+
+    ucs_recursive_spin_lock(&iface->remote_cache.lock);
+
+    khiter = kh_put(cuda_ipc_rem_cache, &iface->remote_cache.hash, pid,
```

Comment:
is kh_put better than kh_get to check if the key is present or not?

Reply:
Using kh_put simplifies the flow here as we don't have to explicitly handle different cases. kh_put can involve resize operation but that's a cost we have to incur and handle at some point even if we are just using kh_get. kh_put seems to additionally check if an element was deleted which is not particularly useful for us other than knowing that the key isn't present. With these tradeoffs I don't know if there is a big difference between the two. @yosefe any suggestions?

Reply:
kh_get is better but not by much, since we indent to create a key if it doesn't exist we could use kh_put for simplicity